### PR TITLE
fix: add missing redis cache service config

### LINKED ISSUE #2486: [NV-1508] 🐛 Bug Report: [ioredis] Unhandled error event: Error: connect ECONNREFUSED 127.0.0.1:6379
### 📜 Description

issue with api service when build with  docker image 
followed  below steps ad facing issue with api application 
# Get the code
git clone --depth 1 https://github.com/novuhq/novu
# Go to the docker folder
cd novu/docker
# Copy the example env file
cp .env.example ./local/deployment/.env
# Start Novu
docker-compose -f ./local/deployment/docker-compose.yml up

### 👟 Reproduction steps

# Get the code
git clone --depth 1 https://github.com/novuhq/novu
# Go to the docker folder
cd novu/docker
# Copy the example env file
cp .env.example ./local/deployment/.env
# Start Novu
docker-compose -f ./local/deployment/docker-compose.yml up

api        | [ioredis] Unhandled error event: Error: connect ECONNREFUSED 127.0.0.1:6379
api        |     at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1278:16)
api        | [ioredis] Unhandled error event: Error: connect ECONNREFUSED 127.0.0.1:6379
api        |     at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1278:16)
api        | [ioredis] Unhandled error event: Error: connect ECONNREFUSED 127.0.0.1:6379
api        |     at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1278:16)

### 👍 Expected behavior

it should connect with redis container as per .env file 

### 👎 Actual Behavior with Screenshots

api        | [ioredis] Unhandled error event: Error: connect ECONNREFUSED 127.0.0.1:6379
api        |     at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1278:16)
api        | [ioredis] Unhandled error event: Error: connect ECONNREFUSED 127.0.0.1:6379
api        |     at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1278:16)
api        | [ioredis] Unhandled error event: Error: connect ECONNREFUSED 127.0.0.1:6379
api        |     at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1278:16)
api        | [ioredis] Unhandled error event: Error: connect ECONNREFUSED 127.0.0.1:6379
api        |     at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1278:16)
api        | [ioredis] Unhandled error event: Error: connect ECONNREFUSED 127.0.0.1:6379
api        |     at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1278:16)

### 🤖 Node Version

_No response_

### 📃 Provide any additional context for the Bug.

_No response_

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/novuhq/novu/blob/main/CONTRIBUTING.md)

### Are you willing to submit PR?

None

<sub>[NV-1508](https://linear.app/novu/issue/NV-1508/🐛-bug-report-[ioredis]-unhandled-error-event-error-connect)</sub>

### PR DESCRIPTION
**Pull request by titenkov**
### What change does this PR introduce?

Fix missing Redis cache service config in `docker-compose.yml`. 

### Why was this change needed?

Closes #2486

### CODE REVIEW

**Comment by scopsy** [comment]
This is very strange, a connection shouldn't be made if this env variable is not specified. Do we have some defaults for it that perhaps we need to remove? @djabarovgeorge

File: docker/local/deployment/docker-compose.yml [hunk 1]
```diff
@@ -34,6 +34,8 @@ services:
       REDIS_HOST: ${REDIS_HOST}
       REDIS_PORT: ${REDIS_PORT}
       REDIS_DB_INDEX: 2
+      REDIS_CACHE_SERVICE_HOST: ${REDIS_HOST}
+      REDIS_CACHE_SERVICE_PORT: ${REDIS_PORT}
```
Comment (scopsy) [inline docker/local/deployment/docker-compose.yml:38]:
```suggestion
      REDIS_CACHE_SERVICE_HOST: ${REDIS_CACHE_SERVICE_HOST}
      REDIS_CACHE_SERVICE_PORT: ${REDIS_CACHE_SERVICE_PORT}
```

Is it possible to separate both service? One is used for the queue and the other for caching functionalitys

**Comment by djabarovgeorge** [comment]
> This is very strange, a connection shouldn't be made if this env variable is not specified. Do we have some defaults for it that perhaps we need to remove? @djabarovgeorge

it did not create a connection therefore issue #2486 mine fault I probably missed it while refactoring it to a separate env variable.

**Comment by titenkov** [comment]
Should it be enabled by default when running locally in `docker-compose` or it should stay disabled?

File: docker/local/deployment/docker-compose.yml (hunk 1 above)
Comment (titenkov) [reply docker/local/deployment/docker-compose.yml]: Done

**Comment by scopsy** [comment]
@titenkov should stay disabled by default for the cache service

**Comment by titenkov** [comment]
> @titenkov should stay disabled by default for the cache service

Done

**Review by scopsy — APPROVED** [review]
I restored some of the configurations for test and dev since we want redis to be enabled there by default

**Pull request merged**